src/abstractions/Text.tsx

```diff
@@ -39,17 +39,25 @@ export const Text = forwardRef(({ anchorX = 'center', anchorY = 'middle', childr
   const [nodes, text] = useMemo(() => {
     let n: React.ReactNode[] = []
     let t = ''
-    Children.forEach(children, (child) => {
+    let key = 0
+    Children.forEach(children, (child, index) => {
       if (typeof child === 'string') {
         t += child
       } else if (child && typeof child === 'object' && (child as React.ReactElement).props.attach === 'material') {
         // Instantiate the base material and grab a reference to it, but don't assign any
         // props, and assign it as the `material`, which Troika will replace behind the scenes.
-        n.push(createElement((child as React.ReactElement).type, { ref: setBaseMtl, attach: 'material' }))
+        n.push(createElement((child as React.ReactElement).type, { ref: setBaseMtl, attach: 'material', key: key++ }))
         // Once the base material has been assigned, grab the resulting upgraded material,
         // and apply the original material props to that.
         if (baseMtl) {
-          n.push(<primitive object={textRef.current.material} {...(child as React.ReactElement).props} attach={null} />)
+          n.push(
+            <primitive
+              object={textRef.current.material}
+              {...(child as React.ReactElement).props}
+              key={key++}
```

Comment:
you can just do key={index} though, no need for the baseMtl, the key is just a unique identifier.

Reply:
any ideas about the conflict? this is because of the changes we made to avoid extends right?

Reply:
@drcmda I believe that because they are both being pushed as the first-level results of a `map`, any `key` must be unique between them. In normal JSX syntax, this is avoided by using a `React.Fragment` which forced a single element which is returned. I didn't verify this just now, but I believe this is the normal behavior for these patterns:

```jsx
// ❗ expects a single child as a return
// ⚠️warning duplicate keys
map((_, i) => (
  <div key={i} />
  <div key={i} />
))

// ⚠️warning duplicate keys
map((_, i) => {
  const results = []
  results.push(<div key={i} />);
  results.push(<div key={i} />);
  return results
})

// ✔️success
map((_, i) => {
  const results = []
  results.push(<div key={`first-${i}`} />);
  results.push(<div key={`second-${i}`} />);
  return results
})

// ✔️success
map((_, i) => (
  <React.Fragment key={i}>
    <div />
    <div />
  </React.Fragment>
))

```

Reply:
> any ideas about the conflict?

Yea, it's probably due to the extends. It's a bit more difficult to resolve conflicts in the github.com interface though. I can look into pushing this ticket through later today though if it isn't resolved by then.